[DX-1012, TT-11584] Improved endpoint cache docs (#4054)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-gateway/middleware/endpoint-cache-tyk-oas.md

---
title: Using the Endpoint Caching middleware with Tyk OAS APIs
date: 2024-01-24
description: "Using the Endpoint Caching middleware with Tyk OAS APIs"
tags: ["Caching", "Cache", "Endpoint Cache", "selective caching", "middleware", "per-endpoint", "Tyk OAS"]
---

The [Endpoint Caching]({{< ref "basic-config-and-security/reduce-latency/caching/advanced-cache" >}}) middleware allows you to perform selective caching for specific endpoints rather than for the entire API, giving you granular control over which paths are cached.

> When working with Tyk OAS APIs the middleware is configured in the [Tyk OAS API Definition]({{< ref "tyk-apis/tyk-gateway-api/oas/x-tyk-oas-doc#operation" >}}) either manually within the `.json` file or from the API Designer in the Tyk Dashboard.

Author: ```suggestion
When working with Tyk OAS APIs the middleware is configured in the [Tyk OAS API Definition]({{< ref "tyk-apis/tyk-gateway-api/oas/x-tyk-oas-doc#operation" >}}). You can do this via the Tyk Dashboard API or in the API Designer.
```

    "x-tyk-api-gateway": {
        "info": {
            "name": "example-endpoint-cache",
            "state": {
                "active": true
            }
        },
        "upstream": {
            "url": "http://httpbin.org/"
        },
        "server": {
            "listenPath": {
                "value": "/example-endpoint-cache/",
                "strip": true
            }
        },
        "global": {
            "cache": {
                "enabled": true,
                "timeout": 60
            }
        },
        "middleware": {
            "operations": {
                "delay/5post": {
                    "cache": {
                        "enabled": true,
                        "cacheResponseCodes": [
                            200
                        ],
                        "timeout": 5
                    }
                }
            }
        }
    }
}
```

In this example the endpoint cache middleware has been configured to cache HTTP 200 responses to HTTP `POST` requests to the `/delay/5` endpoint. The cache will refresh after 5 seconds. Note that requests to other endpoints will also be cached, with a default cache timeout of 60 seconds.

> The configuration above is a complete and valid Tyk OAS API Definition that you can import into Tyk to try out the API-level response header transform.

Reviewer: this confuses me a bit, is this considered part of the response header transform?

## Configuring the middleware in the API Designer
Adding endpoint caching to your API endpoints is easy when using the API Designer in the Tyk Dashboard, simply follow the steps taken in this short video:

>  < placeholder for video >

Reviewer 2: Remember to add placeholder video

Author: Added screenshots for now
